### PR TITLE
Feature/nifti extension writer

### PR DESCRIPTION
**Pull request by twitzelbos**
Add basic extension writing to the nifti crate

### CODE REVIEW

File: src/writer.rs
Comment (samitbasu) [inline src/writer.rs]: Normally, this would be something like `with_extensions` to follow the builder pattern.

File: src/extension.rs
Comment (samitbasu) [inline src/extension.rs:109]: Does this not take a NiftiECode as an argument for some reason?

File: src/writer.rs
Comment (samitbasu) [inline src/writer.rs:273]: Since you had to look this up, a comment to the effect that the vox offset needs to be a float32 per the Nifti spec would help future maintainers.

File: src/writer.rs
Comment (samitbasu) [inline src/writer.rs:323]: Why not just merge these two into one function that takes an `Option<ExtensionSequence>`?  Then you can reduce the number of calls everywhere else by removing the `write_header_terminator` call.

File: src/extension.rs
Comment (twitzelbos) [reply src/extension.rs:109]: Well, if I did that, then we would have to make QBIO private codes part of this repo. I added NiftiECode to represent the few reserved ecodes as a nice helper. If you have a suggestion on how to allow custom/private ecodes and requiring NiftiECode to be the type, I can do something.

File: src/writer.rs
Comment (twitzelbos) [reply src/writer.rs]: Done

File: src/writer.rs
Comment (twitzelbos) [reply src/writer.rs:273]: Done, but did it in the structure definition

File: src/writer.rs
Comment (twitzelbos) [reply src/writer.rs:323]: Done with `Option<&ExtensionSequence>`

**Review by samitbasu — COMMENTED** [review]
I don't understand why `with_extensions` would take an `Option`.  If you have no extensions to add, then you won't call it.

**Comment by twitzelbos** [comment]
> 

Because I'm a C idiot. Fixed.

**Comment by twitzelbos** [comment]
Oops, wrong button

**Comment by twitzelbos** [comment]
@samitbasu pretty please?

File: src/extension.rs
Comment (samitbasu) [reply src/extension.rs:109]: It's fine for now.  You _could_ simply make these `const` values in the crate and leave it to the end user to do validation.  That's how I handle the mini header in our code.

**Review by samitbasu — APPROVED** [review]

**Pull request merged**
